examples/structured_data/classification_with_tfdf.py

```diff
@@ -0,0 +1,726 @@
+            raise ValueError(
+                f"You need to call the adapt method to compute target encoding statistics."
+            )
+
+        inputs = self._standardize_inputs(inputs)
+
+        # Fix and validate inputs shape.
+        if inputs.shape.rank == 1:
+            inputs = tf.expand_dims(inputs, axis=-1)
+        elif inputs.shape.rank > 2:
+            raise ValueError(
+                f"inputs is expected to be of rank 2, got {data.shape.rank}"
+            )
+        # Validate the dimensions of the second axis.
+        if inputs.shape[-1] != 1:
+            raise ValueError(
+                f"The second axis of inputs is expected to have 1 dimension, got {data.shape[-1]}"
+            )
+        # Lookup positive frequencies for the input feature values.
+        positive_fequency = tf.cast(
+            tf.gather_nd(self.positive_frequency_lookup, inputs),
+            dtype=tf.dtypes.float32,
+        )
+        # Lookup negative frequencies for the input feature values.
+        negative_fequency = tf.cast(
+            tf.gather_nd(self.negative_frequency_lookup, inputs),
+            dtype=tf.dtypes.float32,
+        )
+        # Compute positive probability for the input feature values.
+        positive_probability = positive_fequency / (
+            positive_fequency + negative_fequency
+        )
+        # Concatenate and return the looked-up statistics.
+        return layers.concatenate(
+            [positive_fequency, negative_fequency, positive_probability], axis=1
+        )
+
+    def compute_output_shape(self, input_shape):
+        # The output shape is expected to be [num_examplesm 3].
+        return tf.TensorShape([input_shape[0], 3])
+
+    def compute_output_signature(self, input_spec):
+        output_shape = self.compute_output_shape(input_spec.shape.as_list())
+        return tf.TensorSpec(shape=output_shape, dtype=tf.int64)
+
+    def _standardize_inputs(self, inputs):
+        inputs = tf.convert_to_tensor(inputs)
+        if inputs.dtype != tf.dtypes.int64:
+            inputs = tf.cast(inputs, tf.dtypes.int64)
+        return inputs
+
+
+"""
+Let's test the binary target encoder
+"""
+
+data = tf.constant(
+    [
+        [0, 1],
+        [2, 0],
+        [0, 1],
+        [1, 1],
+        [1, 1],
+        [2, 0],
+        [1, 0],
+        [0, 1],
+        [2, 1],
+        [1, 0],
+        [0, 1],
+        [2, 0],
+        [0, 1],
+        [1, 1],
+        [1, 1],
+        [2, 0],
+        [1, 0],
+        [0, 1],
+        [2, 0],
+    ]
+)
+
+binary_target_encoder = BinaryTargetEncoding()
+binary_target_encoder.adapt(data)
+print(binary_target_encoder([0, 1, 2]))
+
+"""
+### Implement a feature encoding with target encoding
+"""
+
+
+def create_target_encoder():
+    inputs = create_model_inputs()
+    target_values = train_data[[TARGET_FEATURE_NAME]].to_numpy()
+    encoded_features = []
+    for feature_name in inputs:
+        if feature_name in CATEGORICAL_FEATURES_WITH_VOCABULARY:
```

Comment:
Target mean should also be used on categorical feature stored as integers.

Reply:
`CATEGORICAL_FEATURES_WITH_VOCABULARY` include both categorical features with string values and integer values.